kubernetes-exec-helper/src/Kubernetes/CreateWSClient.hs

```diff
@@ -0,0 +1,137 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Kubernetes.CreateWSClient 
+  (
+    CreateWSClient
+    , writer
+    , channels
+    , container
+    , kubernetesConfig
+    , commands
+    , getTimeOut
+    , createWSClient
+    , clusterClientParams
+    , getHeaders
+    , getConnectionOptions
+    , prompt
+    , getContainer
+    , getHost
+    , getPort
+    , getURIAuth
+  )
+where 
+import Control.Concurrent.STM
+import Data.ByteString.Lazy as LazyBString
+import Data.ByteString.Lazy.Internal as LByteString 
+import Data.ByteString as S
+import Data.Maybe
+import Data.Monoid ((<>))
+import Data.Text 
+import Data.Text.Encoding as E (encodeUtf8)
+import Network.Socket as Socket
+import Network.HTTP.Types.URI
+import Network.TLS as TLS (ClientParams(..))
+import Network.WebSockets as WS (Headers, defaultConnectionOptions, ConnectionOptions)
+import Network.URI
+import Text.Printf as Printf 
+import Kubernetes.K8SChannel as K8SChannel
+import Kubernetes.Model
+import Kubernetes.KubeConfig
+import Kubernetes.Core
+import Kubernetes.Client
+
+newtype Route = Route {_route :: String}
+{- | 
+  CreateWSClient maintains all threads that are running,  
+  a writer channel to send messages to the server 
+  and a list of all "(ChannelId, TChan Text)" pairs.
+-}
+data CreateWSClient a = CreateWSClient {
+    _writer :: TChan a -- ^ Write back to the server.
+    , _channels :: [(ChannelId, TChan a)] -- ^ Read from the server
+    , _kubernetesConfig :: KubernetesConfig
+    , _clusterClientParams :: TLS.ClientParams
+    , _container :: V1Container -- ^ The container
+    , _commands :: [Command] -- ^ the list of commands to run.
+    , _timeOutInterval :: Maybe TimeoutInterval
+    }
+
+prompt :: Text 
+prompt = "k8:s>"
+
+getHeaders :: CreateWSClient a -> WS.Headers 
+getHeaders _ = [] 
+
+writer :: CreateWSClient a -> TChan a 
+writer clientState = _writer clientState
+
+channels :: CreateWSClient a -> [(ChannelId, TChan a)]
+channels aState = _channels aState
+
+container :: CreateWSClient a -> V1Container 
+container = _container
+
+kubernetesConfig :: CreateWSClient a -> KubernetesConfig
+kubernetesConfig = _kubernetesConfig
+
+commands :: CreateWSClient a -> [Command]
+commands client = _commands client
+
+clusterClientParams :: CreateWSClient a -> TLS.ClientParams 
+clusterClientParams client = _clusterClientParams client 
+
+getTimeOut :: CreateWSClient a -> Maybe TimeoutInterval 
+getTimeOut client = _timeOutInterval client
```

Comment:
All these one-line functions seem quite unnecessary to me. Am I missing something?